File: src/1684.count-the-number-of-consistent-strings.1786/solution.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
    let result = 0;
    const set = new Set(allowed);
    for (const word of words) {
        if (word.split('').every(letter => set.has(letter))) {
            result++;
        }
    }
    return result;
};

File: src/1684.count-the-number-of-consistent-strings.1786/solution.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int result = 0;
        boolean[] contains = new boolean[26];
        for (char c : allowed.toCharArray()) {
            contains[c - 'a'] = true;
        }
        for (String word : words) {
            boolean flag = true;
            for (char c : word.toCharArray()) {
                if (!contains[c - 'a']) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                result++;
            }
        }

        return result;
    }
}